language/move-lang/tests/move_check/parser/spec_parsing_ok.move

```diff
@@ -0,0 +1,101 @@
+// These are some basic parsing tests for specifications which are expected to succeed.
+// Full testing of spec parsing correctness is done outside of this crate.
+//
+// Note that even though we only test parsing, we still need to ensure that the move code (not the specification)
+// is type checking correctly, because with no parsing error, the test harness
+// will run subsequent phases of the move-lang compiler.
+//
+// For parse failures, see the `spec_*_fail.move` test cases.
+
+module M {
+    synthetic expected_coin_sum: u64;
+    synthetic other: bool;
```

Comment:
What are these? I'm a bit confused by their name/usage

Reply:
Specification variables. They are now in a spec block, so the name is not longer needed.